### PR TITLE
Retry namespace update for TTL labels

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

Still no canonical patch generation library, so this will do

/cc @smarterclayton

### CODE REVIEW

**Comment by smarterclayton** [comment]
Should only be retrying on conflict.

**Comment by smarterclayton** [comment]
and we have a helper for it that does this for you in the client library.

File: cmd/ci-operator/main.go [hunk 1]
```diff
@@ -590,7 +590,16 @@ func (o *options) initializeNamespace() error {
 		for key, value := range updates {
 			update.ObjectMeta.Annotations[key] = value
 		}
-		if _, err := client.Namespaces().Update(update); err != nil {
+		var updateErr error
+		for retries := 5; retries > 0; retries-- {
+			if _, err := client.Namespaces().Update(update); err == nil {
```
Comment (smarterclayton) [inline cmd/ci-operator/main.go]: this won't work, you have to actually re fetch and reapply your changes.

Comment (stevekuznetsov) [reply cmd/ci-operator/main.go]: Oh ugh, we should just use patch. Why is that hard today with client libraries?

**Pull request merged**